proxy/http/HttpSM.cc

```diff
@@ -4062,13 +4062,17 @@ HttpSM::do_remap_request(bool run_inline)
   }
 
   // check if the overridden client cert filename is already attached to an existing ssl context
-  ats_scoped_str clientCert(Layout::relative_to(t_state.txn_conf->client_cert_filepath, t_state.txn_conf->client_cert_filename));
-  auto tCTX = params->getCTX(clientCert);
-
-  if (tCTX == nullptr) {
-    // make new client ctx and add it to the ctx list
-    auto tctx = params->getNewCTX(clientCert);
-    params->InsertCTX(clientCert, tctx);
+  if (t_state.txn_conf->client_cert_filepath && t_state.txn_conf->client_cert_filename) {
```

Comment:
Are these set if the cert name is `NULL`? We shouldn't be doing the `ats_scoped_str` if this is guaranteed to fail. In fact, the string allocation should certainly be moved under the check on `clientCert` otherwise a useless string allocation is done every remap.